frontend/packages/operator-lifecycle-manager/src/components/operand.tsx

```diff
@@ -163,8 +163,61 @@ export const OperandTableHeader = () => {
   ];
 };
 
+export enum OperatorStatusType {
+  phase = 'phase',
+  status = 'status',
+  state = 'state',
+  conditions = 'conditions',
+}
+
+export const OperatorStatusTypeText = {
+  [OperatorStatusType.phase]: 'Phase:',
+  [OperatorStatusType.status]: 'Status:',
+  [OperatorStatusType.state]: 'State:',
+  [OperatorStatusType.conditions]: 'Condition:',
+};
+
+export const OperandStatusIconAndText: React.FunctionComponent<OperandStatusIconAndTextProps> = ({
+  statusObject,
+}) => {
+  let iconAndText = <div className="text-muted">Unknown</div>;
+  if (_.isEmpty(statusObject)) {
+    return iconAndText;
+  }
+  _.forEach(OperatorStatusType, (key: string) => {
+    if (_.has(statusObject, key)) {
+      const status = statusObject[key];
+      if (!_.isEmpty(status)) {
+        let statusIcon;
```

Comment:
I think we can simplify this a lot. I would avoid trying to check if the icon exists and just find the first value that's specified (phase/status/state/condition), then pass it to `Status` as-is. If we really want a default icon, we should have a `defaultIcon` prop to the `Status` component. This way we don't need the `StatusType` enum and some of the logic here.

Reply:
I'm inclined to think this is a bit hard to follow yet as well.
Suggestion would be to extract the layer of `statusIcon` out into a separate func.